### ISSUE TITLE
[Design System Toolbox] Add the light mode / dark mode swap interaction

### ISSUE THREAD

**Issue by B3nz01d** [Orange-OpenSource/ouds-ios#944]
## User Story
As a user of the Design Toolbox app
I want to be able to change the mode of the app from within the app
So that:
* I can switch from light to dark mode
* I can switch from dark to light mode
* I can set it to follow the OS setup

## Context
The target implementation can be reviewed on the flutter implementation.

The User should find a button on the top app bar to the far right.

This button should be able to be set by the user upon tap from:
* light mode
* dark mode
* automatic
* back to light mode again

The button should reflect the current state of the setup. When the icon shows the light mode icon then the current screen should be in light mode.

Upon first opening of the app, the screen should be set to the current OS mode.

The icons used are:
* light: ui-light-mode
* dark: ui-dark-mode
* automatic: theme-system

**Comment by B3nz01d** [comment]
This functionnality would add :
* the ability for the user to perform the switch within the app without having to change the entire phone's settings
* the awarness of the Design system implementation covering the support of these various modes 

I am concious that you are proposing the component in both modes within one screen but it does not have the same effect to the user when the entire screen has been switched from one mode to the other then when the component itself is shown in both.

This will add percieved value, but I do understand it does not need to be dealt with as a high priority

**Comment by pylapp** [comment]
📣 **New TestFlight ALPHA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox ALPHA (944)**
- Version: **0.19.0**
- Build number: **1758725245**
- Build tag: **1df3e17**


cc @B3nz01d (product owner)
cc @pylapp @ludovic35 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (a11y team)

_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.20.0**
- Build number: **1759180718**
- Build tag: **be4937f**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.20.0**
- Build number: **1759873602**
- Build tag: **a7cd7f9**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.20.0**
- Build number: **1760476448**
- Build tag: **a8255a9**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
> [!NOTE]
> Find attached the video of the evolution

https://github.com/user-attachments/assets/0a7a35b1-459f-41e4-9768-bb26d911e793

**Comment by B3nz01d** [comment]
closed during sprint # 22
